src/main/java/amazin/controller/BookController.java

```diff
@@ -97,48 +97,23 @@ public String deleteBook(@PathVariable("id") Long id, RedirectAttributes attribu
         return createRedirectViewPath(REQUEST_MAPPING_BOOK);
     }
 
-    @GetMapping("/fuzzySearchName")
-    public String fuzzySearchByBookName(@RequestParam(value = "keywords", required = false) String text, Model model) {
-        List<Book> searchResults = null;
-        try {
-            searchResults = searchservice.fuzzySearch(text, "name");
-
-        } catch (Exception ex) {
-            // for now do nothing. Later we will throw an exception and display a message to
-            // the user
-        }
-        model.addAttribute(MODEL_ATTRIBUTE_BOOK, searchResults);
-        return createRedirectViewPath(REQUEST_MAPPING_BOOK);
-    }
-
-    @GetMapping("/keywordSearchName")
-    public String keywordSearchByBookName(@RequestParam(value = "keywords", required = false) String text,
-            Model model) {
-        List<Book> searchResults = null;
-        try {
-            searchResults = searchservice.keywordSearch(text, "name");
+    @GetMapping("/searchByName")
+    public String searchByBookName(@RequestParam(value = "keywords", required = true) String text, Model model) {
 
-        } catch (Exception ex) {
-            // for now do nothing. Later we will throw an exception and display a message to
-            // the user
-        }
-        model.addAttribute(MODEL_ATTRIBUTE_BOOK, searchResults);
-        return createRedirectViewPath(REQUEST_MAPPING_BOOK);
-    }
+        if (text.isEmpty())
+            return createRedirectViewPath(REQUEST_MAPPING_BOOK);
 
-    @GetMapping("/wildcardSearchName")
-    public String wildcardSearchByBookName(@RequestParam(value = "keywords", required = false) String text,
-            Model model) {
         List<Book> searchResults = null;
         try {
-            searchResults = searchservice.wildcardSearch(text, "name");
-
+            searchResults = searchservice.fuzzySearch(text, "name");
         } catch (Exception ex) {
+            ex.printStackTrace();
             // for now do nothing. Later we will throw an exception and display a message to
```

Comment:
Future cases like this could this could be tagged with a comment like /** TODO handle this exception properly and display message to user */

Most IDE's have support to locate TODO's and let you know/warn you of existing TODO's in the code.

Reply:
TODO has been added